src/main/webapp/app/assessment-shared/assessment-header/assessment-header.component.ts

```diff
@@ -33,4 +35,11 @@ export class AssessmentHeaderComponent {
     @Output() cancel = new EventEmitter<void>();
     @Output() resolveConflict = new EventEmitter<void>();
     @Output() nextSubmission = new EventEmitter<void>();
+
+    constructor(private accountService: AccountService, private route: ActivatedRoute) {}
+
+    ngOnInit() {
+        this.isAtLeastInstructor = this.accountService.hasAnyAuthorityDirect(['ROLE_ADMIN', 'ROLE_INSTRUCTOR']);
+        this.showBackButton = this.route.snapshot.queryParamMap.get('showBackButton') === 'true';
```

Comment:
So to recap how this works:
- The router is used to access a component that has the `assessment-layout` embedded somewhere in it.
- Then the `assessment-header` component within the `assessment-layout` uses the url of the route to extract the showBackButton.

I think this is weird, because a client should only have to look at the top level component (here `assessment-layout`) to understand the configuration options. As mentioned above it should be and input or avoided as it seems to be true for all clients atm.

Reply:
Refactored this.